### PR TITLE
[Scala] Add finch

### PR DESCRIPTION
**Pull request by waghanza**
Hi @whiplash,

This is a relocation of you PR  #3643.

Regards,

------
Closes #3643

### CODE REVIEW

File: scala/finch/src/main/scala/yolo/Main.scala [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  }
+
+  val postUser = post("user") {
+    okEmpty
+  }
+
+  val getUserName = get("user" :: path[String]) { name: String => Ok(name) }
+
+  val api: Service[Request, Response] =
+    Bootstrap
```
Comment (sergeykolbasov) [inline scala/finch/src/main/scala/yolo/Main.scala:25]: I believe most users construct a single endpoint before bootstrapping it:

```scala
Bootstrap
      .serve[Text.Plain](postUser :+: getUserName :+: root)
      .toService
```

**Review by whiplash — DISMISSED** [review]

**Review by whiplash — APPROVED** [review]

File: scala/finch/project/build.properties [hunk 2]
```diff
@@ -0,0 +1 @@
+sbt.version=1.4.4
```
Comment (waghanza) [inline scala/finch/project/build.properties:1]: do we really need this file @whiplash ?

File: scala/finch/project/plugins.sbt [hunk 3]
```diff
@@ -0,0 +1 @@
+addSbtPlugin("com.typesafe.sbt" % "sbt-native-packager" % "1.7.3")
```
Comment (waghanza) [inline scala/finch/project/plugins.sbt:1]: can we use a less strict constraint ? I mean the promise here, is to ensure having a specific version of `finch`, we probably do not need to specify version of `sbt-native-packager` (we can use the latest, I think) ?


the think is to have the same build tool for each `scala` based frameworks

File: scala/finch/project/build.properties (hunk 2 above)
Comment (whiplash) [reply scala/finch/project/build.properties:1]: It can be 1.4.6 (see https://github.com/the-benchmarker/web-frameworks/blob/master/scala/Dockerfile#L1)

File: scala/finch/project/plugins.sbt (hunk 3 above)
Comment (whiplash) [reply scala/finch/project/plugins.sbt:1]: Latest is now 1.8.0... we can possibly use the usual "range" thing I guess. can you make that change?

File: scala/finch/project/build.properties (hunk 2 above)
Comment (waghanza) [reply scala/finch/project/build.properties:1]: can not it be removed ? I mean the docker image contains already an `sbt` version

File: scala/finch/project/plugins.sbt (hunk 3 above)
Comment (waghanza) [reply scala/finch/project/plugins.sbt:1]: my question was more to avoid pin this dependency
for example, by using http://ant.apache.org/ivy/history/latest-milestone/ivyfile/dependency.html#revision

Comment (whiplash) [reply scala/finch/project/plugins.sbt:1]: and I replied that I agree... we can set that plugin version something like `[1.7,2.0]`

Comment (waghanza) [reply scala/finch/project/plugins.sbt:1]: let me rephrase : is there a way to remove this version ? or tell **sbt** to use the last available (not depending on a range) ?

Comment (whiplash) [reply scala/finch/project/plugins.sbt:1]: by reading [here](https://ant.apache.org/ivy/history/2.3.0/ivyfile/dependency.html#revision) I think we can "try" with `latest.integration`

Comment (waghanza) [reply scala/finch/project/plugins.sbt]: seems 🆗

Comment (waghanza) [reply scala/finch/project/plugins.sbt]: but `sbt` warns about
```
[warn] There may be incompatibilities among your library dependencies; run 'evicted' to see detailed eviction warnings.
```
